src/data-workspace/indicators-table-body/use-indicator-value.js

```diff
@@ -3,43 +3,90 @@ import { useMemo, useState, useEffect } from 'react'
 import { useForm } from 'react-final-form'
 import { useBlurredField } from '../../shared/index.js'
 
+// Use a mathematical expression parser instead of `eval`
 const parser = new Parser()
 const evaluate = (expression) => parser.parse(expression).evaluate()
-const formulaPattern = /#\{.+?\}/g
-const separator = '.'
 
-const getDataElementTotalValue = (dataElementCocs) => {
-    if (!dataElementCocs || typeof dataElementCocs !== 'object') {
+/*
+ * In an indicator expression template an `id` can be one of:
+ * 1. #{deId} for a data element
+ * 2. #{deId.cocId} for a category option combo
```

Comment:
Wouldn't this be the `dataElementOperand`? Eg. the `categoryOptionCombo` for a particular `dataElement`.